docs-md/contributing_guide.md

```diff
@@ -0,0 +1,275 @@
+
+```mermaid
+flowchart LR
+    Clone[Clone Repository] --> Setup[Setup Environment]
+    Setup --> Install[Install Dependencies]
+    Install --> Dev[Start Development]
+    
+    subgraph "Environment Setup Steps"
+        Clone --> GitRepo[Github Repository]
+        Setup --> VirtualEnv[Python Virtual Environment]
+        Install --> DevDeps[Development Dependencies]
+        Install --> WebUI[Web UI Dependencies]
+    end
+```
+
+1. **Clone the repository**:
+
+   ```bash
+   git clone https://github.com/locustio/locust.git
+   cd locust
+   ```
+
+2. **Setup the environment using uv**:
+
+   ```bash
+   # Install uv if you don't already have it
+   pip install uv
+   
+   # Create a virtual environment 
+   uv venv
+  
+   # Activate the virtual environment
+   # On Unix/macOS:
+   source .venv/bin/activate
+   # On Windows:
+   .\.venv\Scripts\activate
+   
+   # Install the project with all dependency groups (build, test, lint)
+   uv sync
+   
+   # If you only want specific dependency groups
+   # uv sync --group build --group test
+   ```
+
+   Alternatively, you can use Hatch directly (also used in the project):ł
+
+   ```bash
+   # Install hatch if you don't have it
+   pip install hatch
+   
+   # Create a virtual environment and run tests
+   hatch run test:all
+   ```
+
+3. **Install Web UI dependencies** (if working on UI changes):
+
+   ```bash
+   cd locust/webui
+   yarn install
+   ```
+
+   For more detailed information on Web UI development, including setup, testing, and best practices, see the [Developer Guide](developer_guide.md#web-ui-development).
+
+## Contribution Workflow
+
+```mermaid
+flowchart TD
+    Issue[Find or Create Issue] --> Branch[Create Branch]
+    Branch --> Develop[Make Changes]
+    Develop --> Test[Run Tests]
+    Test -->|Passes| PR[Create Pull Request]
+    Test -->|Fails| Develop
+    PR --> Review[Code Review]
+    Review -->|Changes Requested| Develop
+    Review -->|Approved| Merge[Merged to Main]
+```
+
+1. **Find or create an issue**:
+   - Check existing issues on the GitHub repository
+   - Create a new issue if needed, clearly describing the bug or feature
+
+2. **Create a branch**:
+
+   ```bash
+   git checkout -b feature/your-feature-name
+   # or
+   git checkout -b fix/bug-description
+   ```
+
+3. **Make your changes**:
+   - Follow the coding standards (see below)
+   - Implement your feature or fix
+   - Add/update tests as appropriate
+
+4. **Run tests**:
+
+   ```bash
+   pytest
+   ```
+
+5. **Create a pull request**:
+   - Push your branch to GitHub
+   - Create a pull request with a clear description
+   - Link the relevant issue(s)
+
+6. **Code review**:
+   - Address any feedback from reviewers
+   - Make requested changes and push updates
+
+7. **Merge**:
+   - Once approved, your PR will be merged into the main branch
+
+## Coding Standards
+
+Locust follows these coding standards:
+
+1. **Code Formatting**:
+   - Python code is formatted with Black
```

Comment:
In [2.21.0](https://github.com/locustio/locust/releases/tag/2.21.0) black was migrated to Ruff